src/algebra/m231.rs

```diff
@@ -94,6 +95,40 @@ impl From<u32> for Mod231 {
     }
 }
 
+impl From<i32> for Mod231 {
+    fn from(x: i32) -> Self {
+        let y: u32 = if x < 0 {
+            let y = x + MODULUSI32;
+            if y < 0 {
+                panic!("i32 out of range for Mod231")
```

Comment:
FWIW, `From` should not be fallible. If we permit for the possibility of generating an error when converting from one type to another, we should use [TryFrom] trait.

[TryFrom]: https://doc.rust-lang.org/std/convert/trait.TryFrom.html

Reply:
Done, have alook. Eventually, conversion from `u32` should use `TryFrom` as well, but this is potentially a breaking change.